griptape/drivers/sql/snowflake_sql_driver.py

```diff
@@ -15,24 +15,26 @@
 
 @define
 class SnowflakeSqlDriver(BaseSqlDriver):
-    connection_func: Callable[[], SnowflakeConnection] = field(kw_only=True)
+    connection_provider: Callable[[], SnowflakeConnection] = field(kw_only=True)
```

Comment:
Might be nice to create some reusable types for these to make it easier for customer code create their own abstractions.

For example adding a:

```py
ConnectionProvider = Callable[[], SnowflakeConnection]
```

In this file, then using it here:


```py
connection_provider: ConnectionProvider = field(kw_only=True)
```

And you can reduce duplicate in `def validate_connection_provider(` too.

I can also see arguments against this, but I like being able to easily references types in my own code.

Reply:
yeah I like the idea, but that can be introduced in the future.